src/Microsoft.Identity.Web.TokenCache/TokenCacheExtensions.cs

```diff
@@ -19,7 +19,8 @@ namespace Microsoft.Identity.Web
     /// </summary>
     public static class TokenCacheExtensions
     {
-        private static readonly ConcurrentDictionary<MethodInfo, IServiceProvider> s_serviceProviderFromAction
+        //internal for testing only
+        internal static readonly ConcurrentDictionary<MethodInfo, IServiceProvider> s_serviceProviderFromAction
```

Comment:
The MSAL tests need a way to reset this cache because when they run in a batch, they have the potential to collide and tests fail. Enabling tests to reset this fixed this issue.

Reply:
CC @jennyf19 @jmprieur